### ISSUE TITLE
Duplicated reblog count on i18n FormattedMessage

### ISSUE THREAD

**Issue by xatier** [mastodon/mastodon#32614]
### Steps to reproduce the problem

Since https://github.com/mastodon/mastodon/pull/29585, the UI shows duplicated reblog/favorite counts.

Simply open up a regular toot and see the reblog/favorate box.

![Image](https://github.com/user-attachments/assets/2eee036a-24e9-46fc-b136-34601ccf254e)

### Expected behaviour

Having only one counter from the rendered UI

### Actual behaviour

Seeing the counters been duplicated.

### Detailed description

It's a convention that people on Crowdin translate the i18n string `{count, plural, one {boost} other {boosts}}` into `{count, plural, other {# 則轉嘟}}`, where `#` will be substituted with the value of `count`. Crowdin also provides the preview UI with the counter to see the substituted strings.

![Image](https://github.com/user-attachments/assets/2b3b6be4-7bec-4e38-920e-251ea50dd218)

Since the following code been introduced, both the `<AnimatedNumber>` tag and `<FormattedMessage>` will render the `reblogs_count` value if the translators follow the convention, which results in duplicated counts in the UI.

We should avoid this pattern in HTML constructs, or should **explicitly** inform the translators on Crowdin to avoid using `#` in the translated strings.

```html
        <Link to={`/@${status.getIn(['account', 'acct'])}/${status.get('id')}/reblogs`} className='detailed-status__link'>
          <span className='detailed-status__reblogs'>
            <AnimatedNumber value={status.get('reblogs_count')} />
          </span>
          <FormattedMessage id='status.reblogs' defaultMessage='{count, plural, one {boost} other {boosts}}' values={{ count: status.get('reblogs_count') }} />
        </Link>
```

### Mastodon instance

g0v.social

### Mastodon version

v4.3.0

### Browser name and version

Edge 131.0.2889.0 (Official build) dev (64-bit) 

### Operating system

Arch linux

**Comment by renchap** [comment]
This is indeed a mistake in the source. We should have the animated counter code be a variable to be included in the translatable string, similar to what we do for `hashtag.counter_by_accounts`, like:

```ts
<FormattedMessage
  id='status.reblogs'
  defaultMessage='{count, plural, one {{counter} boost} other {{counter} boosts}}'
  values={{
    count: status.get('reblogs_count'),
    counter: (
      <span className='detailed-status__reblogs'>
        <AnimatedNumber value={status.get('reblogs_count')} />
      </span>
    ),
  }}
/>
```

This would not allow the translator to use `#`, but you could chose where the rich animated count component appears

**Comment by xatier** [comment]
Thank you for looking into this issue.

**Comment by xatier** [comment]
The PR has been merged.

**Comment by ChaosExAnima** [comment]
reopening, as we're going to need to revert the PR due to unintended side effects. see #36834 and #36833.

edit: examining the translation files, it appears this issue affects the following languages: AR, CY, NAN, SR-LATN, SR, and SV.

it does make sense to include the number in the translated string, but until all the strings are converted this would strip out the counts entirely. i wonder if there's a way to conditionally check, or we can update all languages with the `{counter}` tag.

**Comment by xatier** [comment]
> it does make sense to include the number in the translated string, but until all the strings are converted this would strip out the counts entirely. i wonder if there's a way to conditionally check, or we can update all languages with the `{counter}` tag.

I believe it should be fixed forward in the translation (to use `{counter}`, as I mentioned in the PR. Introducing conditional code would be unnecessary complexity, IMO.

It's a situation where you either have the translation to fit the broken code or the other way around.

As long as we have all translation updated by the next release along with this PR for the code fix, we are not really impacting users (besides those who are running the pre-released verions).

**Comment by xatier** [comment]
A simple (but no-so-correct) fix would be adding a `#` (number) to the translation strings not using `{counter}`, this will at least give the users a number in the UI.

**Comment by ChaosExAnima** [comment]
many servers are running the nightly, so leaving the strings to be updated until the next version isn't a good option. and _no_ languages are using `{counter}` as that was added in the PR.

**Comment by xatier** [comment]
False, at least the zh-tw translation [1] was updated as the PR was merged.

I would leave the decision to the mastadon team.

[1] https://crowdin.com/editor/mastodon/52/en-zhtw?view=comfortable&filter=basic&value=0&search_scope=everything&search_strict=0&search_full_match=0&case_sensitive=0#q=%7Bcount,%20plural,%20other%20%7B%7Bcounter%7D%20
